### DISCUSSION TITLE
Reuse chunks

### DISCUSSION THREAD

**Discussion by FCACollin** [quarto-dev/quarto-cli#1161] [Q&A] [answered]
As a statistical programmer, using R and R Markdown, I was used to reuse code chunks such as explained here: <https://bookdown.org/yihui/rmarkdown-cookbook/reuse-chunks.html>.

````
Here is a code chunk that is not evaluated:

```{r, chunk-one, eval=FALSE}
1 + 1
2 + 2
```

Now we actually evaluate it:

```{r, chunk-one, eval=TRUE}
```

And I can evaluate it later again in the same document:

```{r, chunk-one, eval=TRUE}
```

````
I have attempted to find information about this feature for quarto documents, with no success. Do you know if this is implemented somehow in quarto?

Thanks in advance.

**Comment by cderv** [comment, marked as answer]
Everything that works with **knitr** should work with Quarto. 

So this works 
````markdown
---
title: test
format: html
---

Here is a code chunk that is not evaluated:

```{r, chunk-one, eval=FALSE}
1 + 1
2 + 2
```

Now we actually evaluate it:

```{r, chunk-one, eval=TRUE}
```

And I can evaluate it later again in the same document:

```{r, chunk-one, eval=TRUE}
```
````

At least on my side with latest version. 

Is this correctly working for you ?

**Reply by FCACollin** [reply]
Thanks for trying. At least now I know that it should work although it does not work yet on machine... 

I am working on docker image rocker/verse:4.2.0. When I render the example:

![image](https://user-images.githubusercontent.com/51078094/174083652-0927e297-837b-442a-be90-b87e6558573a.png)

I have added the `sessioninfo::session_info()` below. Any idea about what could be the root cause of the problem?

```
─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.2.0 (2022-04-22)
 os       Ubuntu 20.04.4 LTS
 system   x86_64, linux-gnu
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       Etc/UTC
 date     2022-06-16
 pandoc   2.17.1.1 @ /usr/lib/rstudio-server/bin/quarto/bin/ (via rmarkdown)

─ Packages ───────────────────────────────────────────────────────────────────
 package     * version date (UTC) lib source
 cli           3.3.0   2022-04-25 [1] RSPM (R 4.2.0)
 digest        0.6.29  2021-12-01 [1] RSPM (R 4.2.0)
 evaluate      0.15    2022-02-18 [1] RSPM (R 4.2.0)
 fastmap       1.1.0   2021-01-25 [1] RSPM (R 4.2.0)
 htmltools     0.5.2   2021-08-25 [1] RSPM (R 4.2.0)
 htmlwidgets   1.5.4   2021-09-08 [1] RSPM (R 4.2.0)
 jsonlite      1.8.0   2022-02-22 [1] RSPM (R 4.2.0)
 knitr         1.39    2022-04-26 [1] RSPM (R 4.2.0)
 magrittr      2.0.3   2022-03-30 [1] RSPM (R 4.2.0)
 rlang         1.0.2   2022-03-04 [1] RSPM (R 4.2.0)
 rmarkdown     2.14    2022-04-25 [1] RSPM (R 4.2.0)
 rstudioapi    0.13    2020-11-12 [1] RSPM (R 4.2.0)
 sessioninfo   1.2.2   2021-12-06 [1] RSPM (R 4.2.0)
 stringi       1.7.6   2021-11-29 [1] RSPM (R 4.2.0)
 stringr       1.4.0   2019-02-10 [1] RSPM (R 4.2.0)
 xfun          0.31    2022-05-10 [1] RSPM (R 4.2.0)
 yaml          2.3.5   2022-02-21 [1] RSPM (R 4.2.0)

 [1] /usr/local/lib/R/site-library
 [2] /usr/local/lib/R/library
```

**Reply by cderv** [reply]
I think this is because there was an issue in Quarto that is now fixed https://github.com/quarto-dev/quarto-cli/issues/1034
So you should update your quarto version to at least [v0.9.499](https://github.com/quarto-dev/quarto-cli/releases/tag/v0.9.499) but latest would be best

**Reply by FCACollin** [reply]
Thanks for the hint and the news. I confirm:

![image](https://user-images.githubusercontent.com/51078094/174104771-a43a65e5-06ed-458b-be1b-9b44574603b6.png)
